go/arrow/cdata/cdata_test.go

```diff
@@ -769,6 +769,21 @@ func TestExportRecordReaderStream(t *testing.T) {
 	assert.EqualValues(t, len(reclist), i)
 }
 
+func TestExportRecordReaderStreamLifetime(t *testing.T) {
+	reclist := arrdata.Records["primitives"]
+	rdr, _ := array.NewRecordReader(reclist[0].Schema(), reclist)
+
+	out := createTestStreamObj()
+	ExportRecordReader(rdr, out)
+	releaseStream(out)
+
+	// Release once
+	rdr.Release()
+
+	// Should panic (we have no other way of checking reference count)
+	assert.Panics(t, func() { rdr.Release() })
```

Comment:
this is only going to panic when the `assert` tag is present, which could be a problem.

Reply:
Hmm. Is there a better way of testing it?

This isn't a strictly necessary test, just realized we didn't have an explicit test of this.

Reply:
The better way of testing it would be to use the checked allocator and test records that are allocated with the checked allocator with a `defer mem.AssertSize(t, 0)` like we do in other areas

Reply:
Ah, forgot about that. Updated